### PR TITLE
WC Beta E2E tests - Add workaround to revert to classic checkout

### LINKED ISSUE #7599: Update E2E tests to address failures related to changes included in the latest WC Beta
### Describe the bug

With the new [WC 8.3.0-beta-1 released on November 1, 2023](https://developer.woocommerce.com/2023/11/01/woocommerce-8-3-0-beta-1-released/), WC Blocks will be used as default for cart & checkout pages. Since the current E2E tests uses non blocks checkout for most of the flows, they're failing.

To reduce constant failures due to the above change, we've temporarily skipped runningE2E tests against WC Beta versions - https://github.com/Automattic/woocommerce-payments/pull/7598. The beta tests should be added back once the tests are updated.

### To Reproduce

Refer to failing E2E tests - https://github.com/Automattic/woocommerce-payments/actions/runs/6719397754

### PR DESCRIPTION
**Pull request by achyuthajoy**
Fixes #7599

#### Changes proposed in this Pull Request

This PR adds a workaround for E2E test failures when using WC 8.3.0 or above since these versions uses blocks for checkout & cart pages. Though WC 8.3.0 or above uses block layout, it is possible to switch to classic layout by using the `[woocommerce_cart]` & `[woocommerce_checkout]` shortcodes.

This PR updates the setup script to check whether the installed version is >= WC 8.3.0 & if yes, modifies the cart & checkout page to use shortcodes via WP-CLI. 

However, this is a temporary workaround since we have blocks tests for basic payment flows. A permanent solution would be to update the tests to use block checkout flows.

#### Testing instructions

* Against this branch trigger a new [E2E Tests - All ](https://github.com/Automattic/woocommerce-payments/actions/workflows/e2e-test.yml) and confirm that beta tests are enabled again and are passing as expected. (Ref run - https://github.com/Automattic/woocommerce-payments/actions/runs/6823945830)

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7690` or branch name `update/e2e-wc-8.3-workaround` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/e2e-wc-8.3-workaround) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 1015f853e409ba6a7609cf15b58dc03a65b884f5
- Build time: 2023-11-10 11:42:37 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: tests/e2e/env/setup.sh [hunk 1]
```diff
@@ -210,6 +210,23 @@ cli wp option set woocommerce_enable_signup_and_login_from_checkout "yes"
...
+IS_WORKAROUND_REQUIRED=$(cli_debug wp eval "echo version_compare(\"$INSTALLED_WC_VERSION\", \"8.3\", \">=\");")
+
+if [[ "$IS_WORKAROUND_REQUIRED" = "1" ]]; then
+	echo "Updating cart & checkout pages for WC > 8.3 compatibility..."
+	# Get cart & checkout page IDs.
+	CART_PAGE_ID=$(cli_debug wp option get woocommerce_cart_page_id)
+	CHECKOUT_PAGE_ID=$(cli_debug wp option get woocommerce_checkout_page_id)
+
+	# Update cart & checkout pages to use shortcode.
+	cli wp post update "$CART_PAGE_ID" --post_content="<!-- wp:shortcode -->[woocommerce_cart]<!-- /wp:shortcode -->"
```
Comment (deepakpathania) [inline tests/e2e/env/setup.sh]: Could we get the shortcode commands from constant variables as well? Makes the final command look much more readable imo.

**Review by deepakpathania — APPROVED** [review]
Pre approving with a small comment, feel free to merge after addressing.

File: tests/e2e/env/setup.sh (hunk 1 above)
Comment (achyuthajoy) [reply tests/e2e/env/setup.sh]: Yeah sure. Will move them to variables

Comment (achyuthajoy) [reply tests/e2e/env/setup.sh]: Changes added on https://github.com/Automattic/woocommerce-payments/pull/7690/commits/1015f853e409ba6a7609cf15b58dc03a65b884f5

**Pull request merged**